R/get_confidence_set.R

```diff
@@ -32,14 +35,16 @@
 #' )
 #' # get the confidence set
 #' cs_results <- get_confidence_set(
-#'   predictors = cbind(BB_proportions$coaching_updt, BB_proportions$launch_duration,
-#'   BB_proportions$birth_volume_100),
+#'   predictors_data = cbind(
+#'     BB_proportions$coaching_updt, BB_proportions$launch_duration,
+#'     BB_proportions$birth_volume_100
+#'   ),
 #'   fitted_model = model,
 #'   outcome_goal = 0.8,
 #'   outcome_type = "continuous",
 #'   center_characteristic_list = "birth_volume_100",
 #'   center_cha = 1.74,
-#'   outcome = BB_proportions$EBP_proportions,
+#'   outcome_data = BB_proportions$EBP_proportions,
```

Comment:
I noticed that for a lot of this the users will need to specify the column name with the specific data, and it won't compile with just the column name. Perhaps in the instructions above (the comments part at the beginning), we should specify: #' @param outcome_data A `numeric` vector. This should be the column containing the outcome data, extracted directly from the dataset (e.g., `dataset$column_name` or `dataset[["column_name"]]`). Do not pass the column name as a string (e.g., `"column_name"`).

If you agree with this, we might need to create another warning message such as:
if (is.character(outcome_data)) {
    stop("`outcome_data` should be a numeric vector (e.g., dataset$column_name), not a column name as a character string.")
}
for several parameters that require parsing columns, all in one warning message.

Reply:
I think this example might have confused you a bit. this is just an example for testing the `get_confidence_set` function. `get_confidence_set` is not a user facing function, it is being called [here at line 489](https://github.com/correspondMerchant/LAGO-R-Package/pull/5/files#diff-69b091a2ea4ffe095b059062485d7d288fb80d86a379bacea2b114f9ff5f6cbaR489) by `calculate_recommended_interventions`. In my opinion, all the preliminary argument checks should be done in `calculate_recommended_interventions`